docs/guides/computed-properties.mdx

```diff
@@ -6,40 +6,104 @@ description: 'Use object getters and setters'
 
 # Computed Properties
 
-Valtio itself doesn't provide a functionality for computed properties,
-but you can emulate it with object getters and setters.
+In Valtio you can use object & class getters and setters to create computed properties.
 
-⚠️ Note: This is a very advanced usage.
+⚠️ Note: This is advanced usage due to the nuances around `this` behavior; see the "Note about this `this`" section below.
 
-## Simple getter
+## Simple object getter
 
 ```js
 const state = proxy({
   count: 1,
   get doubled() {
-    return state.count * 2
+    return this.count * 2
```

Comment:
we avoid teaching `this` in the first place. only add a note that they could do it.

Reply:
I will revert in my next push, but fwiw if I were to read `return state.count` in a getter in basically any code review, I would immediately assume a) it was a bug, and b) it doesn't work.

Granted, I would be wrong (it does work), b/c JS can actually capture the `const state` and eval `state.count` from within `get doubled()`.

But, dunno, I just really want to push/sanity check that in _any_ language (JS, TS, Java, C#, etc), in _any_ framework (mobx, ember), a getter that does _not_ do `return this.whatever` is a bug and so, dunno, this being Valtio's up-front / recommended usage "looks weird" and I'm worried will cause more confusion than the "avoiding `this`" intention hopes to avoid.

Nonetheless, will revert.